### PR TITLE
Milestone 2 - Decorate class

### PR DESCRIPTION
**Pull request by mukhammadrizooff**
# In this branch I implement:
- I check the usage of [Decorator pattern in Ruby](https://refactoring.guru/design-patterns/decorator/ruby/example#example-0)
- I use two decorators in order to capitalize and trim people's names.
- In the following steps I get instructions for implementing the Decorator design pattern for this case.

#### Interface
- Create a class `Nameable`.
   - Implement a method called `correct_name` that will raise a `NotImplementedError`.

#### Turn your Person class to Nameable
- Make sure that my Person class inherits from Nameable
- Make sure that this class has a method `correct_name` implemented. It should simply return the name attribute.

#### Prepare base Decorator
- Make sure that it inherits from Nameable.
- In the constructor assign a nameable object from params to an instance variable.
- Implement the `correct_name` method that returns the result of the correct_name method of the `@nameable`.

#### Prepare CapitalizeDecorator and TrimmerDecorator
- For the CapitalizeDecorator:
    - Create a class that inherits from the base Decorator class.
    - Implement a method `correct_name` that capitalizes the output of `@nameable.correct_name`.
- For the TrimmerDecorator:
    - Create a class that inherits from the base Decorator class.
    - Implement a method `correct_name` that makes sure that the output of `@nameable.correct_name` has a maximum of 10 characters. If it's longer it should trim the word.

### CODE REVIEW

File: base_decorator.rb [hunk 1]
```diff
@@ -0,0 +1,12 @@
+require_relative 'nameable'
+
+class Decorator < Nameable
+  def initialize(nameable)
+    super
```
Comment (TedLivist) [inline base_decorator.rb]: - [x] There was an error running the tests as a result of this `super`. There is a difference between `super` and `super()`. You can refer to this [Stackoverflow answer](https://stackoverflow.com/a/31816281) for a more detailed explanation. Kindly remove this `super` or change it to `super()` to solve this error.

  rizo-1

File: person.rb [hunk 2]
```diff
@@ -1,4 +1,6 @@
-class Person
+require_relative 'nameable'
+
+class Person < Nameable
   attr_reader :id
   attr_accessor :name, age
```
Comment (TedLivist) [inline person.rb]: - [x] Kindly ensure your syntax is correct. Your Person class threw up errors here. You should have `:age` here (with `:`)

File: person.rb [hunk 3]
```diff
@@ -7,12 +9,17 @@ def initialize(id, age, name = 'Unknown', parent_permission: true)
     @name = name
     @age = age
     @parent_permission = parent_permission
+    super
```
Comment (TedLivist) [inline person.rb]: - [x] There was also an issue running your app as a result of this `super`. Kindly resolve it.

File: capitalize_decorator.rb [hunk 4]
```diff
@@ -0,0 +1,7 @@
+require_relative 'base_decorator'
+
+class CaptalizeDecorator < Decorator
+  def correct_name
+    @nameable.correct_name.upcase
```
Comment (TedLivist) [inline capitalize_decorator.rb]: - [x] You are expected to Capitalize your name, not make it UPPERCASE. Kindly refactor your method to return 'Maximilianus', not 'MAXIMILIANUS' as we have in the image below.

  rizo-2

**Review by TedLivist — CHANGES_REQUESTED** [review]
## STATUS: CHANGES REQUIRED ♻️ ♻️

Hi @mukhammadrizooff,

Good job so far!
✔️ Good documentation
✔️ Classes are in different files
However, there are some issues that you still need to work on to go to the next project but you are almost there!

## Required Changes ♻️
_Check the comments under the review._

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear. And please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Review by t-yanick — APPROVED** [review]
Hi @mukhammadrizooff, 

Your project is complete!  There is nothing else to say other than... it's time to merge it :shipit:  
Congratulations! 🎉 

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Pull request merged**